### PR TITLE
Fibonacci search in Java

### PR DESCRIPTION
**Pull request by iamharshitpandeyy**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

Fibonacci Search in array using Java

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

#3292
kindly review my PR @bhav09 
Thank you.

### CODE REVIEW

File: Java/search/Fibonacci_search.java [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+        {
+            location += 1;
+            System.out.println("The given element " + key + " found at location " + location + " ");
+        }
+
+        // The given key is not present
+        else {
+            System.out.println("The given element " + key + " is not present in the given array");
+        }
+
```
Comment (bhav09) [inline Java/search/Fibonacci_search.java]: Remove redundant lines.

File: Java/search/Fibonacci_search.java [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+   The given element 52 found at location 8
+
+ */
+
+/*
+  COMPLEXITY
+  
+  Time complexity O(log n) 
+ 
+ */
```
Comment (bhav09) [inline Java/search/Fibonacci_search.java]: Add a new line.

**Review by bhav09 — CHANGES_REQUESTED** [review]
Reviewed.

File: Java/search/Fibonacci_search.java [hunk 3]
```diff
@@ -0,0 +1,133 @@
+/*
+1) Find a Fibonacci number that is greater than or equal to the size of the array in which we are searching for the key.
+2) Compare the key with the predecessor of the Fibonacci number obtained in Step 1 and store it in index.
+3) If the key and array element at index are equal then the key is present at position index + 1.        
+4) If the key is less than array element at index, then search the left sub-tree to index.
+5) If the given key is greater than the array element at index, then search the right sub-tree to index.  
+6) If the key is not found, repeat the steps from Step 1 to Step 5 as long as index = 0, that is, Fibonacci number >= array_size. After each iteration the size of array array_size is reduced.
+                
+*/
```
Comment (ankitaggarwal23) [inline Java/search/Fibonacci_search.java]: - Use Multiline comments
- Make it concise

File: Java/search/Fibonacci_search.java [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+
+package Java;
+
+import java.util.*;
```
Comment (ankitaggarwal23) [inline Java/search/Fibonacci_search.java]: Remove the extra spaces

File: Java/search/Fibonacci_search.java [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+            fibk2 = fibk1;
+            fibk1 = fibk;
+        }
+        return fibk2;
+    }
+
+    public static void main(String[] args) {
+
+        // Creating Scanner Object to read input from user
+        Scanner s = new Scanner(System.in);
```
Comment (ankitaggarwal23) [inline Java/search/Fibonacci_search.java]: Remove redundant comments and spaces

File: Java/search/Fibonacci_search.java [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+        int sorted_Array[] = new int[100];
+        int key;
+        int index;
+        int low;
+        int high;
+        int flag;
+        int location;
+        int array_Size;
+
```
Comment (ankitaggarwal23) [inline Java/search/Fibonacci_search.java]: You can initialize it like that as well `int key, index ,low, high, flag, location, array_Size;` saves lots of lines

File: Java/search/Fibonacci_search.java [hunk 7]
```diff
@@ -0,0 +1,133 @@
...
+        System.out.println("Enter the size of the array: ");
+
+        // Reading the size of the array from the user
+        array_Size = s.nextInt();
+
+        // Reading the elements of the array from the user and storing it in variable
+        // sortedArray
```
Comment (ankitaggarwal23) [inline Java/search/Fibonacci_search.java]: Remove redundant comments and spaces

File: Java/search/Fibonacci_search.java [hunk 8]
```diff
@@ -0,0 +1,133 @@
...
+        System.out.println("Enter the elements of the array: ");
+        for (int i = 0; i < array_Size; i++) {
+            sorted_Array[i] = s.nextInt();
+        }
+        // Initialising the values of some variables
```
Comment (ankitaggarwal23) [inline Java/search/Fibonacci_search.java]: Remove redundant comments and spaces

File: Java/search/Fibonacci_search.java [hunk 9]
```diff
@@ -0,0 +1,133 @@
...
+        low = 0;
+        location = -1;
+        high = array_Size - 1;
+        flag = 0;
+
+        // Read the key element to be searched from the user
+        System.out.println("Enter the element which is to be searched: ");
+        // Storing it in element in variable key
```
Comment (ankitaggarwal23) [inline Java/search/Fibonacci_search.java]: Remove redundant comments and spaces

File: Java/search/Fibonacci_search.java [hunk 10]
```diff
@@ -0,0 +1,133 @@
...
+        key = s.nextInt();
+
+        // Closing the scanner Object
```
Comment (ankitaggarwal23) [inline Java/search/Fibonacci_search.java]: Remove redundant comments and spaces

File: Java/search/Fibonacci_search.java [hunk 11]
```diff
@@ -0,0 +1,133 @@
...
+            // key and array element at index are equal
+            if (key == sorted_Array[index + low]) {
+                location = low + index;
+                flag = 1;
+                break;
+            }
+            // key is greater than array element at index
+            else if (key > sorted_Array[index + low])
+
+            { // Updating the lower index
```
Comment (ankitaggarwal23) [inline Java/search/Fibonacci_search.java]: Remove redundant comments and spaces

File: Java/search/Fibonacci_search.java [hunk 12]
```diff
@@ -0,0 +1,133 @@
...
+                low = low + index + 1;
+            } else {
+                // Updating the higher index
+                high = low + index - 1;
+            }
+            // adjusting the size of the array
+            array_Size = high - low + 1;
+
+        }
+        // The given key is present in the array
```
Comment (ankitaggarwal23) [inline Java/search/Fibonacci_search.java]: Remove redundant comments and spaces

File: Java/search/Fibonacci_search.java [hunk 13]
```diff
@@ -0,0 +1,133 @@
...
+        if (flag == 1)
+        // Incrementing location by one to get the position of key as location
+        // represents index of key
```
Comment (ankitaggarwal23) [inline Java/search/Fibonacci_search.java]: Remove redundant comments and spaces

File: Java/search/Fibonacci_search.java [hunk 14]
```diff
@@ -0,0 +1,133 @@
...
+        {
+            location += 1;
+            System.out.println("The given element " + key + " found at location " + location + " ");
+        }
+
+        // The given key is not present
+        else {
```
Comment (ankitaggarwal23) [inline Java/search/Fibonacci_search.java]: Remove redundant comments and spaces

File: Java/search/Fibonacci_search.java [hunk 15]
```diff
@@ -0,0 +1,133 @@
...
+            System.out.println("The given element " + key + " is not present in the given array");
+        }
+
+    }
+
+}
+
```
Comment (ankitaggarwal23) [inline Java/search/Fibonacci_search.java]: Remove spaces

File: Java/search/Fibonacci_search.java (hunk 2 above)
Comment (ankitaggarwal23) [inline Java/search/Fibonacci_search.java]: - Add Time and Space Complexity at the end of the Code

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]

**Comment by iamharshitpandeyy** [comment]
I had Changes and commit, now Check it @bhav09 @ankitaggarwal23

File: Java/search/Fibonacci_search.java [hunk 16]
```diff
@@ -35,99 +28,61 @@ public static int getFibanociNumber(int array_Size) {
...
    Enter the elements of the array: -15 0 9 14 23 35 49 52 65 77 
    Enter the element which is to be searched: 52 
    The given element 52 found at location 8
 
- */
-
-/*
   COMPLEXITY
-  
   Time complexity O(log n)
```
Comment (ankitaggarwal23) [inline Java/search/Fibonacci_search.java]: Add space complexity

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]
//   Find a Fibonacci number that is greater than or equal to the size of the   array in which we are searching for the key.
//   Compare the key with the predecessor of the Fibonacci number obtained in Step   1 and store it in index.
//   If the key and array element at index are equal then the key is present at   position index + 1.
//   If the key is less than array element at index, then search the left sub-tree   to index.
//   If the given key is greater than the array element at index, then search the   right sub-tree to index.
//   If the key is not found, repeat the steps from Step 1 to Step 5 as long as   index = 0, that is, Fibonacci number >= array_size.   
//   After each iteration the size of array array size is reduced.

**Comment by iamharshitpandeyy** [comment]
I had Changes and commit, now Check it @ankitaggarwal23

File: Java/search/Fibonacci_search.java [hunk 17]
```diff
@@ -0,0 +1,85 @@
+// Find a Fibonacci number that is greater than or equal to the size of the array in which we are searching for the key.
+// Compare the key with the predecessor of the Fibonacci number obtained in Step 1 and store it in index.
+// If the key and array element at index are equal then the key is present at position index + 1.        
+// If the key is less than array element at index, then search the left sub-tree to index.
+// If the given key is greater than the array element at index, then search the right sub-tree to index.  
+// If the key is not found, repeat the steps from Step 1 to Step 5 as long as index = 0, that is, Fibonacci number >= array_size. After each iteration the size of array array_size is reduced.
```
Comment (ankitaggarwal23) [inline Java/search/Fibonacci_search.java]: Rest looks good to merge.
Your last point in the comment exceed the range which looks not good. Only change needed, Split your last comment into two parts:
// If the key is not found, repeat the steps from Step 1 to Step 5 as long as index = 0, that is, Fibonacci number >= array_size.
// After each iteration the size of array array size is reduced.

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]

File: Java/search/Fibonacci_search.java (hunk 17 above)
Comment (iamharshitpandeyy) [reply Java/search/Fibonacci_search.java]: ok

**Comment by iamharshitpandeyy** [comment]
I had Changes and commit, now Merge it @ankitaggarwal23

**Review by ankitaggarwal23 — APPROVED** [review]

**Pull request merged**